sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -282,13 +284,13 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper with Join
             // otherwise the pruning will not trigger
             var filterableScan = getFilterableTableScan(l, left)
             if (filterableScan.isDefined && canPruneLeft(joinType) &&
-                hasPartitionPruningFilter(right)) {
+                hasPartitionPruningFilter(right, r)) {
```

Comment:
Can we filter out the predicates which use references on joining key when passing into `hasPartitionPruningFilter`? Then we don't need to change `hasPartitionPruningFilter`, `hasSelectivePredicate`, etc.

Reply:
I couldn't understand how can we do this filtering here, this should be done recursively on  all the filters in `right` plan.

Reply:
Oh, nvm, I thought the filter predicate was passed in. It is not the case here.